src/lualib/Error.ts

```diff
@@ -0,0 +1,93 @@
+Error = class {
+    public static captureStackTrace(targetObject: {}, depth: number): void {
+        targetObject["stack"] = debug.traceback(undefined, depth);
```

Comment:
If we want to keep non-standard extension exposed let's keep the same function signature as in V8. I think it should be possible by walking through getinfo.

Also, `stack` should contain type and message as well.

Reply:
Good point, that was actually my original intention but I wasn't aware that we can actually get the calling function through `debug.getinfo`. So it may make sense to implement the intended behaviour. `debug.getinfo` has the advantage of respecting source maps though.

Reply:
For now I've renamed the function to `__TS__GetErrorStack(constructor: Function)` that actually behaves similarly as `captureStackTrace` should, but simply returns the stack trace up to the `constructor` provided.

Reply:
> Also, stack should contain type and message as well

According to my tests on [TypeScript Playground](https://www.typescriptlang.org/play/#code/FAMwrgdgxgLglgewgAgOYAoCUBvYz-ICGANgKYBOM6AoueQuegEQAWpxxCyF95TmAOgDOMQlADWmYAF9goSLEQoQWXATRYA3DLkrMQA) the stack does not contain type and message.

Reply:
I've been testing on Safari. In Chrome/V8 the stack does include the error name and message before the traceback. I prefer `Error.stack` to only contain the traceback, but we might just as well accept the V8 as ground truth here.

Reply:
However, as the stack property is set in the constructor and not lazily evaluated due to #725, we cannot append the error name and message as they are still unknown here. So for now let's simply go with Safari's interpretation.